core/src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -229,19 +229,22 @@ private void applyDeletedIndices(final ClusterChangedEvent event) {
                 logger.debug("[{}] cleaning index, no longer part of the metadata", index);
             }
             final IndexService idxService = indicesService.indexService(index);
-            final IndexSettings indexSettings;
+            IndexMetaData metaData;
             if (idxService != null) {
-                indexSettings = idxService.getIndexSettings();
+                final IndexSettings indexSettings = idxService.getIndexSettings();
                 deleteIndex(index, "index no longer part of the metadata");
+                sendDeletedEvent(event.state(), index, indexSettings, localNodeId);
+            } else if ((metaData = previousState.metaData().index(index)) != null) {
```

Comment:
fancy pants :) can we just use metaData().hasIndex() and extract the value in the if body?